### PR TITLE
Add unit tests for b-productswithselectors block

### PR DESCRIPTION
**Pull request by Matei-Iordache**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--www-landing-pages--bitdefender.hlx.page/drafts/test-page/
- After: https://DEX-17015--www-landing-pages--bitdefender.hlx.page/drafts/test-page/

### CODE REVIEW

File: _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    beforeEach(() => {
+      b_productswithselectors = document.querySelector('.b-productswithselectors-container');
+    });
+
+    test('check if products have been added to the window.productsListCount', () => {
+        expect(window.productsListCount).toBeTruthy();
+    });
+
+    test('check if the number of servers and mailboxes changes based on number of devices selected', () => {
+      const UsersSelectorBox = b_productswithselectors.querySelectorAll('.selectorBox')[0];
```
Comment (vradulescu-bd) [inline _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js]: variable name should be lowercase camelcase: usersSelectorBox, capital letter is used for class names; pls try to remove **/*.spec.js from .gitignore

File: _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    
+      // Set the target value on the selector
+      UsersSelectorBox.value = '11';
+      // Dispatch the change event
+      UsersSelectorBox.dispatchEvent(new Event('change'));
+
+      const triggerValue = UsersSelectorBox.value;
+      const fileServers1stProd = Math.ceil((Number(triggerValue)) * 0.3);
```
Comment (vradulescu-bd) [inline _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js]: this is hardcoded logic in the test, that means this test will never fail

File: _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js (hunk 1 above)
Comment (Matei-Iordache) [reply _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js]: updated variable name

File: _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js (hunk 2 above)
Comment (Matei-Iordache) [reply _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js]: hardcoded values

File: _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      usersSelectorBox.value = '11';
+      // Dispatch the change event
+      usersSelectorBox.dispatchEvent(new Event('change'));
+
+      const triggerValue = usersSelectorBox.value;
+      const fileServers1stProd = 4;
+      const fileServers2ndProd = 4;
+      const mailboxes = 17;
+
+      const selectors = [
```
Comment (vradulescu-bd) [inline _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js:46]: naming is to general; which selectors ? what is index, what is type ? maybe add more verbosity in your code otherwise seems hard to follow the logic you have written; also it will be great if we could assert the current and discounted prices; If the updated data is uniform ( not depending of any product type ), then testing 1 box is enough; and withing this box, keep simple and clear selectos: discountedPrice, currentPrice, desktopAndLaptops, fileServers, mailboxes

Comment (Matei-Iordache) [reply _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js:46]: updated data is not uniform, 3 boxes need to be tested, as to code verbosity, why do i need to be more verbose, since I test all the selectors ? I don't test prices in this logic anyway

Comment (vradulescu-bd) [reply _src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js:46]: "I don't test prices in this logic anyway": the price is changing also after the initial selection, so we need to test that also
"as to code verbosity, why do i need to be more verbose, since I test all the selectors ?": because those "indexes" and listChildNumber are way to general, actually you are referring to: discountedPrice, currentPrice, desktopAndLaptops, fileServers, mailboxes. The test needs to be clear what what needs to be tested / asserted
